### PR TITLE
Completed the contributor guide with a section to promote JVM Only ex…

### PR DESCRIPTION
**Pull request by aldettinger**
…tension to native #1021

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: docs/modules/ROOT/pages/contributor-guide.adoc [hunk 1]
```diff
@@ -124,3 +124,134 @@ Review the result visually.
...
+        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
+        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    }
+----
+
+7. Create a native test at `integration-tests/${EXT}/src/test/java/org/apache/camel/quarkus/component/${EXT}/it/${EXT}IT.java`
+
+8. Check `extensions/${EXT}/runtime/src/main/resources/META-INF/quarkus-extension.yaml`:
+* Ensure guide is set to `"https://quarkus.io/guides/camel"`
+* Ensure keyword `camel` is present
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/contributor-guide.adoc:218]: I do not think this is necessary. The keywords are not shown anywhere, they just help to find extensions via search on code.quarkus.io https://github.com/quarkusio/code.quarkus.io/blob/21044d84bee2ca82ac3939df00b9d0646a75f614/src/main/frontend/src/code-quarkus/pickers/extensions-picker-helpers.ts#L13 The word `Camel` is already there in the extension name so all our extensions should be findable via `camel` anyway https://github.com/quarkusio/code.quarkus.io/blob/21044d84bee2ca82ac3939df00b9d0646a75f614/src/main/frontend/src/code-quarkus/pickers/extensions-picker-helpers.ts#L12

File: docs/modules/ROOT/pages/contributor-guide.adoc [hunk 2]
```diff
@@ -124,3 +124,134 @@ Review the result visually.
...
+mvn clean install -D skipTests -P format
+----
+
+11. Build the extension:
++
+[source,shell]
+----
+cd "extensions/${EXT}"
+mvn clean install
+----
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/contributor-guide.adoc:243]: I think this is unnecessary if you do install in step 10.

File: docs/modules/ROOT/pages/contributor-guide.adoc [hunk 3]
```diff
@@ -124,3 +124,134 @@ Review the result visually.
...
+
+12. Execute integration tests:
++
+[source,shell]
+----
+cd "../../integration-tests/${EXT}"
+mvn clean verify -P native
+----
+
+13. Now it's time to solve native build issues if any, extend integration tests coverage and perhaps even shifting some treatments
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/contributor-guide.adoc:253]:
```suggestion
13. Now it's time to solve native build issues if any, extend integration tests coverage and perhaps even shifting some tasks
```

**Comment by aldettinger** [comment]
@ppalaga I've commited simultaneously but I will update in anoher commit.

File: docs/modules/ROOT/pages/contributor-guide.adoc (hunk 1 above)
Comment (aldettinger) [reply docs/modules/ROOT/pages/contributor-guide.adoc:218]: I thought there was a plugin preventing the build but I will double-check.

File: docs/modules/ROOT/pages/contributor-guide.adoc (hunk 2 above)
Comment (aldettinger) [reply docs/modules/ROOT/pages/contributor-guide.adoc:243]: Indeed, it's not useful to check tests at this stage.

**Pull request merged**